shopfloor/services/zone_picking.py

```diff
@@ -899,21 +927,33 @@ def _set_destination_location(self, move_line, quantity, confirmation, location)
             )
             return (location_changed, response)
         # destination location set to the scanned one
-        self._write_destination_on_lines(move_line, location)
+
         stock = self._actions_for("stock")
+        move_lines = move_line
+        if package:
+            quantity = None
+            # Handling all move lines from a complete mix pack
+            for _move_line in package.move_line_ids:
+                if _move_line.state != "assigned":
+                    continue
+                _move_line.qty_done = move_line.product_uom_qty
+                move_lines |= _move_line
+        self._write_destination_on_lines(move_lines, location)
+
         try:
-            stock.mark_move_line_as_picked(move_line, quantity, check_user=True)
+            stock.mark_move_line_as_picked(move_lines, quantity, check_user=True)
         except ConcurentWorkOnTransfer as error:
             response = self._response_for_set_line_destination(
                 move_line,
                 message={
                     "message_type": "error",
                     "body": str(error),
                 },
-                qty_done=quantity,
+                # qty_done=quantity,
```

Comment:
Now it never get passed to the method, it should only be passed to the method,
if we are not handling a package